pkg/neco/cmd/reboot_worker.go

```
@@ -192,7 +192,11 @@ func rebootMain() error {
 		})
 		err := power(context.Background(), "restart", addr)
 		if err != nil {
-			return err
+			log.Warn("failed to restart machine", map[string]interface{}{
+				"serial":    m.Spec.Serial,
+				"node":      m.Spec.IPv4[0],
+				log.FnError: err,
+			})
```

By ignoring errors, the whole command exits successfully even when it has failed to restart some machine.
The importance of the failure varies according to the cause of the failure and the intention of the user, so it would be nice to notify the user of the failure via the exit code.
Then the user will be able to judge what to do.

How about remembering the last error and returning it finally.
It is not necessary to keep a slice of errors because the user can investigate the errors from the warning messages.

---

You're right. I fixed it.